test/test_pytools.py

```diff
@@ -283,6 +283,100 @@ def test_make_obj_array_iteration():
 # }}}
 
 
+def test_tag():
+    from pytools.tag import Taggable, Tag, UniqueTag
+
+    # Need a subclass that defines the copy function in order to test.
+    class TaggableWithCopy(Taggable):
+
+        def copy(self, **kwargs):
+            return TaggableWithCopy(kwargs["tags"])
+
+    class UniqueTagChild(UniqueTag):
+        pass
+
+    class UniqueTagChildChild(UniqueTagChild):
+        pass
+
+    class TagChild(Tag):
+        pass
+
+    class UniqueTagChild2(UniqueTag):
+        pass
+
+    tag1 = UniqueTagChild()
+    tag2 = UniqueTagChildChild()
+    tag3 = Tag()
+    tag4 = TagChild()
+    tag5 = UniqueTagChild2()
```

Comment:
Name these something that makes the resulting code more readable (likely encoding some type info).